source/gui/nvdaControls.py

```diff
@@ -391,3 +401,125 @@ def ScrollChildIntoView(self, child: wx.Window) -> None:
 		finally:
 			# ensure child.GetRect is reset properly even if super().ScrollChildIntoView throws an exception
 			child.GetRect = oldChildGetRectFunction
+
+
+class FeatureFlagCombo(wx.Choice):
+	"""Creates a combobox (wx.Choice) with a list of feature flags.
+	"""
+	def __init__(
+			self,
+			parent: wx.Window,
+			keyPath: List[str],
+			conf,
+			translatedOptions: OrderedDict[FeatureFlagValues, str],
+			pos=wx.DefaultPosition,
+			size=wx.DefaultSize,
+			style=0,
+			validator=wx.DefaultValidator,
+			name=wx.ChoiceNameStr,
+	):
+		"""
+		:param parent: The parent window.
+		:param keyPath: The list of keys required to get to the config value.
+		:param conf: The config.conf object.
+		:param translatedOptions: A dictionary of translated options, for each feature flag value. Note, the
+		FeatureFlagValues.default value should not be included in this dictionary. See L{
+		FeatureFlagCombo._setDefaultOptionLabel}.
+		:param pos: The position of the control. Forwarded to wx.Choice
+		:param size: The size of the control. Forwarded to wx.Choice
+		:param style: The style of the control. Forwarded to wx.Choice
+		:param validator: The validator for the control. Forwarded to wx.Choice
+		:param name: The name of the control. Forwarded to wx.Choice
+		"""
+		self._confPath = keyPath
+		self._conf = conf
+		if FeatureFlagValues.DEFAULT in translatedOptions:
+			raise ValueError(
+				f"The translatedOptions dictionary should not contain the key {FeatureFlagValues.DEFAULT!r}"
+				" It will be added automatically. See _setDefaultOptionLabel"
+			)
+		self._translatedOptions = self._createOptionsDict(translatedOptions)
+		choices = list(self._translatedOptions.values())
+		super(FeatureFlagCombo, self).__init__(
+			parent,
+			choices=choices,
+			pos=wx.DefaultPosition,
+			size=wx.DefaultSize,
+			style=0,
+			validator=wx.DefaultValidator,
+			name=wx.ChoiceNameStr,
+		)
+
+		self.SetSelection(self._getChoiceIndex(self._getConfigValue().value))
+		self.defaultValue = self._getConfSpecDefaultValue()
+		"""The default value of the config spec. Not the "behavior of default".
+		This is provided to maintain compatibility with other controls in the
+		advanced settings dialog.
+		"""
+
+	def _getChoiceIndex(self, value: FeatureFlagValues) -> int:
+		return list(self._translatedOptions.keys()).index(value)
+
+	def _getConfSpecDefaultValue(self) -> FeatureFlagValues:
+		defaultValueFromSpec = self._conf.getConfigValidation(self._confPath).default
+		if not isinstance(defaultValueFromSpec, FeatureFlag):
+			raise ValueError(f"Default spec value is not a str, but {type(defaultValueFromSpec)}")

```

Comment:
Something doesn't look quite right here, is this message more accurate?

```suggestion
			raise ValueError(f"Default spec value is not a FeatureFlag, but {type(defaultValueFromSpec)}, value: {defaultValueFromSpec})
```